File: Unidade I/Pilhas/Pilha Estática/Pilha Estática com Java/Pilha - Alteração no Método Imprimir()/Empilhamento.java
public class Empilhamento implements Empilhavel {
    private int ponteirotopo;
    private Object[] dados;

    public Empilhamento(int tamanho){
        ponteirotopo = -1;
        dados = new Object[tamanho];
    }

    //Sobrecarga de construtor para definir o tamanho padrão do array
    public Empilhamento(){
        this(10);
    }

    @Override
    public boolean cheia(){
        return (ponteirotopo == dados.length - 1);
    }

    @Override 
    public boolean vazia(){
        return (ponteirotopo == -1);
    }

    @Override
    public void empilhar(Object dado){
        if(!cheia()){
            ponteirotopo++;
            dados[ponteirotopo] = dado;
        } else {
            System.out.println("Não é possível empilhar. Pilha cheia.");
        }
    }

    @Override 
    public Object desempilhar(){
        Object dadotopo = null;
        if(!vazia()){
            dadotopo = dados[ponteirotopo];
            ponteirotopo--;
        } else {
            System.out.println("Não é possível desempilhar. Pilha vazia");
        }
        return dadotopo;
    }

    @Override
    public void imprimir(){
        for(int i = 0; i <= ponteirotopo; i++){
            if(i < ponteirotopo){
                System.out.print("[" + dados[i] + "], ");
            } else if (i == ponteirotopo) {
                System.out.println("[" + dados[i] + "]");
            }
        }
    }

    @Override
    public Object topo(){
        Object dadotopo = null;
        if(!vazia()){
            dadotopo = dados[ponteirotopo];
        } else {
            System.out.println("Não existe dado no topo. A pilha está vazia.");
        }
        return dadotopo;
    }


}

File: Pilha Estática/Pilha com Kotlin/Pilha/Empilhamento.kt
class Empilhamento(private var tamanho: Int = 10) : Empilhavel {
    private var ponteirotopo: Int = -1
    private var dados: Array<Any?> = arrayOfNulls(tamanho)

    constructor() : this(10) // Sobrecarga de construtor para tamanho padrão

    override fun cheia(): Boolean {
        return ponteirotopo == dados.size - 1
    }

    override fun vazia(): Boolean {
        return ponteirotopo == -1
    }

    override fun empilhar(dado: Any) {
        if (!cheia()) {
            ponteirotopo++
            dados[ponteirotopo] = dado
        } else {
            println("Não é possível empilhar. Pilha cheia.")
        }
    }

    override fun desempilhar(): Any? {
        var dadotopo: Any? = null
        if (!vazia()) {
            dadotopo = dados[ponteirotopo]
            ponteirotopo--
        } else {
            println("Não é possível desempilhar. Pilha vazia")
        }
        return dadotopo
    }

    override fun imprimir() {
        for (i in 0..ponteirotopo) {
            if (i < ponteirotopo) {
                print("[" + dados[i] + "], ")
            } else if (i == ponteirotopo) {
                println("[" + dados[i] + "]")
            }
        }
    }

    override fun topo(): Any? {
        var dadotopo: Any? = null
        if (!vazia()) {
            dadotopo = dados[ponteirotopo]
        } else {
            println("Não existe dado no topo. A pilha está vazia.")
        }
        return dadotopo
    }
}